tests/parser/features/test_units.py

```diff
@@ -0,0 +1,168 @@
+from decimal import Decimal
+
+from vyper.exceptions import TypeMismatchException, EventDeclarationException
+
+
+def test_function_with_units(get_contract_with_gas_estimation):
+    code = """
+units: {
+    N: "Newton",
+    m: "Meter",
+    s: "Second",
+}
+@public
+def foo(f: uint256(N), d: uint256(m), t: uint256(s)) -> uint256(N*m/s**2):
+    return f * d / (t * t)
+
+@public
+def bar(a: uint256(m), b: uint256(m), c: uint256(m)) -> uint256(m**3):
+    return (a * b * c)
+    """
+
+    c = get_contract_with_gas_estimation(code)
+
+    assert c._classic_contract.abi == [{
```

Comment:
Checking the entire ABI like this is probably fragile. I would suggest checking subsets of the ABI by breaking it out. In particular, gas values might change. No reason to assert that.